router/main_test.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+)
+
+func Test_makeHandler(t *testing.T) {
+	config := []struct {
+		Title              string
+		RequestHost        string
+		RequestURI         string
+		ExpectedSuffix     string
+		UpstreamURL        string
+		Timeout            time.Duration
+		ExpectedStatusCode int
+	}{
+		{
+			Title:              "Right configuration with status 500 unrelated to function",
+			RequestHost:        "martindekov.example.xyz/",
+			RequestURI:         "myfunction",
+			ExpectedSuffix:     "martindekov-myfunction",
+			UpstreamURL:        "http://localhost:8080/",
+			Timeout:            time.Second * 30,
+			ExpectedStatusCode: http.StatusInternalServerError,
+		},
+		{
+			Title:              "Everything configured but upstream URL resulting in status 503",
+			RequestHost:        "martindekov.example.xyz/",
+			RequestURI:         "myfunction",
+			ExpectedSuffix:     "martindekov-myfunction",
+			UpstreamURL:        "",
+			Timeout:            time.Second * 30,
+			ExpectedStatusCode: http.StatusServiceUnavailable,
+		},
+		{
+			Title:              "Everything configured but RequestHost resulting in status 400",
+			RequestHost:        "",
+			RequestURI:         "/myfunction",
+			ExpectedSuffix:     "-myfunction",
```

Comment:
I just wonder if it's simpler for us to have this like:

```
Title
inputURL
expectedUpstreamURL
```

And to leave it at that?